Sources/MarathonCore/Create.swift

```diff
@@ -38,25 +38,74 @@ extension CreateError: PrintableError {
 
 internal final class CreateTask: Task, Executable {
     private typealias Error = CreateError
-
+    
+    private var shouldOpen: Bool {
+        return !argumentsContainNoOpenFlag
+    }
+    
     func execute() throws {
         guard let path = arguments.first?.asScriptPath() else {
             throw Error.missingName
         }
-
+        
+        if arguments.contains("--tests") {
+            if let scriptFile = try? FileSystem().currentFolder.file(atPath: path),
+                let script = try? scriptManager.script(at: scriptFile.path) {
+                try createScriptTests(for: script)
+                try beginEditing(script, if: shouldOpen)
+            } else {
+                let file = try createScript(at: path)
+                let script = try scriptManager.script(at: file.path)
+                try createScriptTests(for: script)
+                try beginEditing(script, if: shouldOpen)
+            }
+        } else {
+            let file = try createScript(at: path)
```

Comment:
Hmmm there should be a way to reduce these code paths. What we want is basically:

1. Create the script file
2. Create the tests if the `--tests` flag has been passed.

Shouldn't this be able to be expressed like this:

```swift
let script = try createScript()

if arguments.contains("--tests") {
    try createTests()
}
```

Maybe I'm over-simplifying, but basically looking for one of the operations to just run on top of the other, instead of having multiple nested `if/else` statements. What do you think @clayellis?